aspnetcore/signalr/streaming.md

```diff
@@ -13,57 +13,107 @@ uid: signalr/streaming
 
 By [Brennan Conroy](https://github.com/BrennanConroy)
 
+::: moniker range="< aspnetcore-3.0"
+
 ASP.NET Core SignalR supports streaming return values of server methods. This is useful for scenarios where fragments of data will come in over time. When a return value is streamed to the client, each fragment is sent to the client as soon as it becomes available, rather than waiting for all the data to become available.
 
-[View or download sample code](https://github.com/aspnet/Docs/tree/live/aspnetcore/signalr/streaming/sample) ([how to download](xref:index#how-to-download-a-sample))
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
 
-## Set up the hub
+ASP.NET Core SignalR supports streaming from client-to-server and from server-to-client. This is useful for scenarios where fragments of data will come in over time. When streaming, each fragment is sent to the client or server as soon as it becomes available, rather than waiting for all the data to become available.
+
+::: moniker-end
+
+::: moniker range="= aspnetcore-2.1"
```

Comment:
The doc will be easier to maintain if you use the following link and display it for all versions:
https://github.com/aspnet/Docs/tree/live/aspnetcore/signalr/streaming

Reply:
I don't know what you mean. The sample has changed every release, shouldn't the user be able to see the correct sample based on what moniker they are looking at?